src/main/kotlin/uk/gov/justice/digital/hmpps/courthearingeventreceiver/config/web/CourtHearingEventFieldsFilter.kt

```diff
@@ -0,0 +1,98 @@
+package uk.gov.justice.digital.hmpps.courthearingeventreceiver.config.web
+
+import com.jayway.jsonpath.Configuration
+import com.jayway.jsonpath.JsonPath
+import com.jayway.jsonpath.PathNotFoundException
+import com.jayway.jsonpath.ReadContext
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.courthearingeventreceiver.service.TelemetryEventType
+import uk.gov.justice.digital.hmpps.courthearingeventreceiver.service.TelemetryService
+import java.util.stream.Collectors
+import javax.servlet.Filter
+import javax.servlet.FilterChain
+import javax.servlet.ServletRequest
+import javax.servlet.ServletResponse
+import javax.servlet.http.HttpServletRequest
+
+@Component
+class CourtHearingEventFieldsFilter(
+  @Autowired private val telemetryService: TelemetryService,
+  @Autowired
+  private val observedFields: ObserveFields
+) : Filter {
+
+  override fun doFilter(request: ServletRequest?, response: ServletResponse?, filterChain: FilterChain?) {
+
+    val httpRequest = request as HttpServletRequest
+    if (observedFields.fields?.isNotEmpty()!!) {
+      val requestWrapper = CustomHttpRequestWrapper(httpRequest)
+      filterChain?.doFilter(requestWrapper, response)
+      val jsonContext: ReadContext = JsonPath.parse(requestWrapper.inputStream)
+      try {
+        trackEvent(jsonContext.jsonString(), observedFields)
+      } catch (exception: UnsupportedOperationException) {
+        return
+      }
```

Comment:
Could check the body present instead but not sure how? 🤔

Reply:
`HttpMethod.POST.matches(httpRequest.method)` should get you part of the way there though you probably will have to keep the exception handling. Probably worth logging a warning if it does catch something.